docs/polygon_draw.js

```diff
@@ -366,22 +387,79 @@ function setUpPolygonDrawing(map, firebasePromise) {
  * @param {google.maps.Map} map Map to display regions on
  * @param {Promise<any>} firebasePromise Promise with Firebase damage data (also
  *     implies that authentication is complete)
- * @return {Promise} promise that is resolved when all initialization is done
- *     (only used by tests)
+ * @return {?google.maps.drawing.DrawingManager} DrawingManager added to map
+ *     (only used by tests). Null if there was an error
  */
-function initializeAndProcessUserRegions(map, firebasePromise) {
+async function initializeAndProcessUserRegions(map, firebasePromise) {
   addLoadingElement(mapContainerId);
-  return firebasePromise
-      .then((doc) => {
-        // Damage asset may not exist yet, so this is undefined. We tolerate
-        // gracefully.
-        damageAsset = doc.data()['asset_data']['damage_asset_path'];
-        userShapes = getFirestoreRoot().collection(collectionName);
-      })
-      .then(() => userShapes.get())
-      .then(
-          (querySnapshot) => drawRegionsFromFirestoreQuery(querySnapshot, map))
-      .catch(createError('getting user-drawn regions'));
+  try {
+    setUpPopup();
+    let doc;
+    try {
+      doc = await firebasePromise;
+    } catch (err) {
+      // Firebase retrieval error handled elsewhere.
+      throw err;
+    }
+    // Damage asset may not exist yet, so this is undefined. We tolerate
+    // gracefully.
+    damageAsset = doc.data()['asset_data']['damage_asset_path'];
+    userShapes = getFirestoreRoot().collection(collectionName);
+    let querySnapshot;
+    try {
+      querySnapshot = await userShapes.get();
+    } catch (err) {
+      handleUserShapesError(err);
+      return null;
+    }
+    drawRegionsFromFirestoreQuery(querySnapshot, map);
+    return setUpPolygonDrawing(map);
+  } finally {
+    loadingElementFinished(mapContainerId);
+  }
+}
+
+const AUTHENTICATION_ERROR_CODE = 'permission-denied';
+const USER_FEATURES_DIALOG =
+    '<div>Sign in to authorized account to view user-drawn features</div>';
+
+/**
+ * Handles errors thrown from retrieving user-defined features, most likely due
+ * to authentication issues.
+ * @param {Error|firebase.FirebaseError} err
+ */
+function handleUserShapesError(err) {
+  const userFeaturesRow = $('#' + userFeaturesCheckboxRowId);
+  userFeaturesRow.css('text-decoration', 'line-through');
+  // Checkbox just causes issues with double event triggering, and is useless
+  // anyway. Get rid of it.
+  userFeaturesRow.children('input').remove();
```

Comment:
I think this looks a little weird. Possible to disable?

Reply:
I wasn't able to get events on a disabled input to fire. https://stackoverflow.com/questions/3100319/event-on-a-disabled-input is similar to what I ran into, but it didn't work for me on Chrome either. That leads to the confusing situation where clicking on the checkbox does nothing, but clicking on the label brings up the dialog. Since I think users are likely to click on the checkbox if it's there, this seemed better.

Tried just making it not disabled. How's that?